### PR TITLE
EventSubscription uses channel:string column instead of receive:boolean

### PR DESCRIPTION
**Pull request by evanrolfe**
The notifications page now uses a drop down to select between 'disabled' and 'instant_email'. Effectively the same as checking/unchecking a checkbox but this allows us to add in a third option (daily_email) which will happen in the next PR.

Here is what the notifications page looks like after this change:
screenshot from 2017-06-21 13-32-01

### CODE REVIEW

**Comment by hennevogel** [comment]
Firefox: 

screenshot from 2017-06-21 14-59-40

**Comment by evanrolfe** [comment]
This is how the page looks for me in firefox, is it possible you need to clear the browser cache or maybe restart the rails server?

screenshot from 2017-06-21 14-05-26

File: src/api/app/models/event_subscription.rb [hunk 1]
```diff
@@ -1,9 +1,14 @@
...
+  CHANNELS = ['disabled', 'instant_email']
+
+  attribute :channel, :string, default: 'disabled'
+
   belongs_to :user, inverse_of: :event_subscriptions
   belongs_to :group, inverse_of: :event_subscriptions
 
   validates :receiver_role, inclusion: { in: [:all, :maintainer, :bugowner, :reader, :source_maintainer,
                                               :target_maintainer, :reviewer, :commenter, :creator] }
+  validates :channel, inclusion: { in: CHANNELS }
```
Comment (hennevogel) [inline src/api/app/models/event_subscription.rb]: enum?

Comment (evanrolfe) [inline src/api/app/models/event_subscription.rb]: @hennevogel done.

**Comment by evanrolfe** [comment]
[(8) P9: Notification frequency [3]](https://trello.com/c/UtWz4k0g/404-8-p9-notification-frequency-3)

**Pull request merged**